core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -699,9 +703,10 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
       // the key address instead of storing the absolute address of the value, the key and value
       // must be stored in the same memory page.
       // (8 byte key length) (key) (value) (8 byte pointer to next value)
-      final long recordLength = 8 + klen + vlen + 8;
+      int uaoSize = UnsafeAlignedOffset.getUaoSize();
+      final long recordLength = (2 * uaoSize) + klen + vlen + 8;
```

Comment:
No, the "8" here has to do with extra padding introduced in 2.0.  The first "8" that was replaced by (2 * uaoSize) has to do with the alignment.